src/main/java/com/notably/model/suggestion/SuggestionModelImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.notably.model.suggestion;
+
+import java.util.List;
+import java.util.Optional;
+
+import javafx.beans.property.Property;
+import javafx.collections.ObservableList;
+
+/**
+ * The implementation class of SuggestionModel.
+ */
+public class SuggestionModelImpl implements SuggestionModel {
+    private ObservableList<SuggestionItem> suggestions;
+    private Property<Optional<String>> responseTextProperty;
+
+    public SuggestionModelImpl(ObservableList<SuggestionItem> suggestions,
+                               Property<Optional<String>> responseTextProperty) {
+        this.suggestions = suggestions;
+        this.responseTextProperty = responseTextProperty;
+    }
+
+    @Override
+    public ObservableList<SuggestionItem> getSuggestions() {
+        return suggestions;
+    }
+
+    @Override
+    public void setSuggestions(List<SuggestionItem> suggestions) {
+        this.suggestions.setAll(suggestions);
+    }
+
+    @Override
+    public Property<Optional<String>> responseTextProperty() {
+        return responseTextProperty;
+    }
+
+    @Override
+    public void setResponseTextProperty(String responseText) {
```

Comment:
Actually, rather than silently accepting null values of `responseText`, I feel like it's better to throw an error if `responseText` is null. Then, you just need to do `Optional.of()`